test/test_pytree.py

```diff
@@ -163,24 +163,42 @@ def test_treemap(self):
         def run_test(pytree):
             def f(x):
                 return x * 3
-            sm1 = sum(map(tree_flatten(pytree)[0], f))
+            sm1 = list(map(f, tree_flatten(pytree)[0]))
             sm2 = tree_flatten(tree_map(f, pytree))[0]
             self.assertEqual(sm1, sm2)
 
             def invf(x):
                 return x // 3
+            sm3 = list(map(f, tree_flatten(pytree)[0]))
+            sm4 = tree_flatten(tree_map(f, pytree))[0]
+            self.assertEqual(sm3, sm4)
```

Comment:
The idea for lines this part replaces was to apply `f` followed by `invf` to get the original pytree.

```python
self.assertEqual(tree_map(invf, tree_map(f, pytree)), pytree)
```

The new lines are currently testing the same thing that `self.assertEqual(sm1, sm2)` is testing.

Reply:
> The idea for lines this part replaces was to apply f followed by invf to get the original pytree.

Interesting. Thanks for sharing, @kshitij12345! Updated the changes.